### PR TITLE
Include sceptre-file-resolver as a core resolver

### PR DESCRIPTION
**Pull request by zxiiro**
This includes the sceptre-file-resolver
(https://github.com/Sceptre/sceptre-file-resolver) into Sceptre as
a core resolver.

Signed-off-by: Thanh Ha <zxiiro@gmail.com>

## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [x] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [x] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [x] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by zxiiro** [comment]
@zaro0508 not sure what you meant by replacing the !file_contents resolver but here I replaced it with the code from the !file resolver. Considering the 2 plugins are pretty similar with the latter being more advanced I would consider this an upgrade to the file_contents resolver.

I think for the most part this should just work for people already using the !file_contents resolver but maybe the parsing of JSON/YAML parsing will come at a surprise to those using this plugin.

File: sceptre/resolvers/file.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+import os
+import json
+import yaml
+import requests
```
Comment (zaro0508) [inline sceptre/resolvers/file.py:6]: is requests in the python 3.6 standard library?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
To add a new plugin we also need to add an entrypoint in `setup.py` and we will want to move the docs to `docs/_source/docs/resolvers.rst`.   Also we want to deprecate the  `!file_contents` resolver and tell everyone to use this new resolver instead.  Could you please add some documentation for that?

File: sceptre/resolvers/file.py (hunk 1 above)
Comment (zxiiro) [reply sceptre/resolvers/file.py:6]: No it's a dependency.

**Comment by zxiiro** [comment]
> To add a new plugin we also need to add an entrypoint in `setup.py` and we will want to move the docs to `docs/_source/docs/resolvers.rst`. Also we want to deprecate the `!file_contents` resolver and tell everyone to use this new resolver instead. Could you please add some documentation for that?

Sure, I'll look into that.

**Comment by craighurley** [comment]
Is the functionality of this existing `!file_contents` resolver identical to a subset of of funcationality in this PRs `!file` resolver?  If so, should this PR be treated as an upgrade to the existing `!file_contents` resolver?  Then there would be no need to deprecate `!file_contents`.

**Comment by zxiiro** [comment]
> Is the functionality of the existing `!file_contents` resolver identical to a subset of of funcationality in this PRs `!file` resolver? If so, should this PR be treated as an upgrade to the existing `!file_contents` resolver? Then there would be no need to deprecate `!file_contents`.

My first implementation in this PR was to upgrade to the functionality of the existing `!file_contents` resolver but I changed it to introduce a new resolver due to @zaro0508's request. I'm happy to re-patch this PR in either direction if the 2 of you can get together and hash it out and get an agreement on which direction we go.

**Comment by zxiiro** [comment]
This has been updated to include the file-resolver by installing it as part of requirements.txt rather than copying the code over from the other repo. I'm not sure what we want to do about the docs. I think keeping it in this repo's docs makes it more official so people can read about it directly in the documentation. If we instead link to the README it feels less official to me. Thoughts?

**Comment by zaro0508** [comment]
i prefer a doc reference otherwise doc changes would need to happen in two repos, correct?

**Comment by zxiiro** [comment]
Updated to reference the GitHub repo for the sceptre-file-resolver for the documentation.

File: setup.py [hunk 2]
```diff
@@ -28,6 +28,7 @@ def get_version(rel_path):
     "Jinja2>=2.8,<3",
     "colorama>=0.3.9",
     "packaging>=16.8,<17.0",
+    "sceptre-file-resolver>=1.0.1,<2",
```
Comment (zaro0508) [inline setup.py:31]: latest version is v1.0.3  https://github.com/Sceptre/sceptre-file-resolver/tags

**Review by zaro0508 — CHANGES_REQUESTED** [review]

**Comment by zaro0508** [comment]
@zxiiro did you forget to push your latest changes?  the diff still shows doc and code additions instead of just making the resolver a dependency.

**Comment by zxiiro** [comment]
> @zxiiro did you forget to push your latest changes? the diff still shows doc and code additions instead of just making the resolver a dependency.

Indeed you are right. I'm not sure how that happened. I just pushed my code up.

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**